Makefile

```diff
@@ -307,30 +310,39 @@ functional-with-fault-injection-test: clean-test-results
 
 test: unit-test integration-test functional-test functional-with-fault-injection-test
 
-##### Coverage #####
+##### Coverage & Reporting #####
 $(COVER_ROOT):
 	@mkdir -p $(COVER_ROOT)
 
-unit-test-coverage: $(COVER_ROOT)
+$(REPORT_ROOT):
+	@mkdir -p $(REPORT_ROOT)
+
+prepare-coverage-test: update-gotestsum $(COVER_ROOT) $(REPORT_ROOT)
+
+unit-test-coverage: prepare-coverage-test
 	@printf $(COLOR) "Run unit tests with coverage..."
-	@echo "mode: atomic" > $(UNIT_COVER_PROFILE)
-	@go test ./$(UNIT_TEST_DIRS) -timeout=$(TEST_TIMEOUT) -race $(TEST_TAG) -coverprofile=$(UNIT_COVER_PROFILE) || exit 1;
+	gotestsum --junitfile $(NEW_REPORT) -- \
```

Comment:
I don't care that much (since I don't use these targets locally), but I was wondering if it would be possible to keep the old `go test` commands _unless_ running in buildkite

Reply:
I was going in that direction at first, but then decided (a) it wasn't worth the trouble and (b) it's better to not have CI-specific switches in the Makefile file. What makes you want to keep using `go test`? As you probably know, all `gotestsum` does is invoke `go test -json` and emit an extra file at the end. So it's really just a thin wrapper that doesn't change the actual test execution.

Reply:
Fair enough, it's a small thing to install. I guess one thought is that if you do use those targets it's going to leave new files each time and they'll just pile up. But I guess the coverage ones will too.. I didn't realize that.

I think it's fine though, you can just run clean once in a while.

Reply:
👍 Yup, that's why I added the extra cleanup instrutions to the `clean-test-results` target. In practice, I believe hardly anyone runs them locally.